mmhuman3d/data/data_converters/lsp.py

```diff
@@ -72,16 +72,17 @@ def convert_by_mode(self, dataset_path: str, out_path: str,
             keypoints2d14 = np.hstack([keypoints2d14, np.ones([14, 1])])
 
             # bbox
-            bbox_xywh = [
+            bbox_xyxy = [
                 min(keypoints2d14[:, 0]),
                 min(keypoints2d14[:, 1]),
                 max(keypoints2d14[:, 0]),
                 max(keypoints2d14[:, 1])
             ]
 
-            if 0 <= bbox_xywh[0] <= w and 0 <= bbox_xywh[2] <= w and \
-                    0 <= bbox_xywh[1] <= h and 0 <= bbox_xywh[3] <= h:
-                bbox_xywh = self._bbox_expand(bbox_xywh, scale_factor=1.2)
+            if 0 <= bbox_xyxy[0] <= w and 0 <= bbox_xyxy[2] <= w and \
+                    0 <= bbox_xyxy[1] <= h and 0 <= bbox_xyxy[3] <= h:
+                bbox_xyxy = self._bbox_expand(bbox_xyxy, scale_factor=1.2)
+                bbox_xywh = self._xyxy2xywh(bbox_xyxy)
```

Comment:
lsp discards example if bbox is outside the image. should the same be done to other datasets? or modify the bbox so it stays within the image?

Reply:
I think since we have the xyxy of expanded bbox `bbox_xyxy = self._bbox_expand(bbox_xyxy, scale_factor=1.2)`, we can check if the expanded bbox falls outside the image and modifies/ clip to ensure it stays within the image. However, this can only be done if h and w is provided in annotation or read from image (which is currently not done in most data converters)

Reply:
Noted, let's stick to the current solution for now.